### PR TITLE
Fix edged log issue

### LINKED ISSUE #2226: kubectl logs return extra linebreak
**What happened**:

**What you expected to happen**:

Only return one byte data.

**How to reproduce it (as minimally and precisely as possible)**:

- run `hack/local-up-kubeedge.sh`
- apply a nginx deployment
- run `kubectl logs --limit-bytes=1` to get nginx log

**Anything else we need to know?**:

The root cause is that we always add linebreak for data whatever it has, see code [here](https://github.com/kubeedge/kubeedge/blob/master/pkg/stream/edgedlogconnection.go#L103).

**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubectl --version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2020-01-14T00:09:19Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}

```

</details>

I use master HEAD to test, v1.4.0 still fails.

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The root cause is that we always add linebreak for data whatever it has, see code [here](https://github.com/kubeedge/kubeedge/blob/master/pkg/stream/edgedlogconnection.go#L103).

**Which issue(s) this PR fixes**:

Fixes #2226

### CODE REVIEW

File: pkg/stream/edgedlogconnection.go [hunk 1]
```diff
@@ -33,6 +34,10 @@ type EdgedLogsConnection struct {
 	ReadChan chan *Message `json:"-"`
 }
 
+var (
+	eol = []byte{'\n'}
```
Comment (fisherxu) [inline pkg/stream/edgedlogconnection.go:38]: Delete the unused variable

File: pkg/stream/edgedlogconnection.go [hunk 2]
```diff
@@ -68,7 +73,7 @@ func (l *EdgedLogsConnection) Serve(tunnel SafeWriteTunneler) error {
 		return err
 	}
 	defer resp.Body.Close()
-	scan := bufio.NewScanner(resp.Body)
+	scan := bufio.NewReader(resp.Body)
```
Comment (fisherxu) [inline pkg/stream/edgedlogconnection.go:76]:
```suggestion
	reader := bufio.NewReader(resp.Body)
```

File: pkg/stream/edgedlogconnection.go [hunk 3]
```diff
@@ -92,21 +97,33 @@ func (l *EdgedLogsConnection) Serve(tunnel SafeWriteTunneler) error {
...
+		}
+		msg := NewMessage(l.MessID, MessageTypeData, data[:n])
+
+		err = msg.WriteTo(tunnel)
 		if err != nil {
 			klog.Errorf("write tunnel message %v error", msg)
 			return err
 		}
-		klog.Infof("%v write logs %v", l.String(), string(scan.Bytes()))
+		klog.Infof("%v write logs %v", l.String(), string(data))
```
Comment (fisherxu) [inline pkg/stream/edgedlogconnection.go:126]: V(4)?

Comment (daixiang0) [reply pkg/stream/edgedlogconnection.go:126]: Good idea.

**Pull request merged**